packages/serverless/src/integrations/aws_lambda.ts

```diff
@@ -0,0 +1,235 @@
+import { EventProcessor, Hub, Integration, Scope } from '@sentry/types';
+import * as Sentry from '@sentry/node';
+
+/**
+ * NodeJS integration
+ *
+ * Provides a mechanism for NodeJS
+ * that raises an exception for handled, unhandled, timeout and similar times of error
+ * and captures the same in Sentry Dashboard
+ */
+export class AWSLambda implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'AWSLambda';
+
+  /**
+   * @inheritDoc
+   */
+  public name: string = AWSLambda.id;
+
+  /**
+   * Returns current hub
+   */
+  private getCurrentHub?: () => Hub;
+
+  /**
+   * context.
+   */
+  private awsContext: any;
+
+  /**
+   * timeout flag.
+   */
+  private timeoutWarning: boolean = false;
+
+  /**
+   * configured time in miliseconds.
+   */
+  private configuredTimeInMilliseconds: number;
+
+  /**
+   * @inheritDoc
+   */
+  public constructor(
+    options: { awslambda?: any; context?: any; timeoutWarning?: any; configuredTimeInMilliseconds?: number } = {},
+  ) {
+    this.awsContext = options.context;
+    this.timeoutWarning = options.timeoutWarning;
+    this.configuredTimeInMilliseconds = options.configuredTimeInMilliseconds as number;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(_: (callback: EventProcessor) => void, getCurrentHub: () => Hub): void {
+    // lambda-bootstrap
+    const lambdaBootstrap = require.main;
+
+    if (!this.awsContext && !lambdaBootstrap) {
+      return;
+    }
+
+    const nodeJs: string = 'node';
+
+    /** capturing this context */
+    const setupOnceContext = this;
+
+    /** rapid runtime instance */
+    const rapidRuntime = lambdaBootstrap?.children[0].exports;
+
+    /** handler that is invoked in case of unhandled and handled exception */
+    const originalPostInvocationError = rapidRuntime.prototype.postInvocationError;
+
+    this.getCurrentHub = getCurrentHub;
+
+    const hub = this.getCurrentHub && this.getCurrentHub();
+
+    /** configured time */
+    this.configuredTimeInMilliseconds = this.awsContext.getRemainingTimeInMillis();
+
+    const timeoutWarningBuffer: number = 1500;
+    const millisToSeconds: number = 1000;
+
+    /** checking if configured Time In Milliseconds is greater than timeout Warning Buffer */
+    if (this.configuredTimeInMilliseconds > timeoutWarningBuffer) {
+      const configuredTimeInSec = Math.floor(this.configuredTimeInMilliseconds / millisToSeconds);
+      const configuredTimeInMilli = Math.floor(configuredTimeInSec * millisToSeconds);
```

Comment:
Why do we even calculate this? We already do have `configuredTimeInMilliseconds`? Both of these options are redundant.
If you want to round `getRemainingTimeInMillis` (which beats the purpose of providing it through the option in the first place), you can just do `Math.floor(getRemainingTimeInMillis / 1000)`. There's no point of having intermediate seconds value.

Reply:
made this change in the new pull request.